tests/test_frame.py

```diff
@@ -2087,6 +2087,10 @@ def test_dict_items() -> None:
     x = {"a": [1]}
     check(assert_type(pd.DataFrame(x.items()), pd.DataFrame), pd.DataFrame)
 
+def test_iteritems() -> None:
+     df = pd.DataFrame({1: [1, 2], None: 5}, columns=pd.Index([1, None], dtype=object))
+     assert_type(df.iteritems(), Iterable[tuple[Hashable, pd.Series]] )
```

Comment:
Probably needs to be in a string for python 3.8: `"Iterable[tuple[Hashable, pd.Series]]"`